modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetPage.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+import {ApiHelpers} from '../../../helpers/ApiHelpers';
+import {ApplicationsMenuPage} from '../../../pages/product-navigation-applications-menu/ApplicationsMenuPage';
+
+export class DataSetPage {
+	readonly apiHelpers: ApiHelpers;
+	readonly applicationsMenuPage: ApplicationsMenuPage;
+	readonly basePath: string;
+	readonly dataSetsTable: Locator;
+	readonly newDataSetButton: Locator;
+	readonly newDataSetModal: {
+		readonly heading: Locator;
+		readonly nameInput: Locator;
+		readonly restApplicationField: Locator;
+		readonly restApplicationOptions: Locator;
+		readonly restEndpointField: Locator;
+		readonly restEndpointOptions: Locator;
+		readonly restSchemaField: Locator;
+		readonly restSchemaOptions: Locator;
+		readonly saveButton: Locator;
+	};
+	readonly page: Page;
+
+	constructor(page: Page) {
+		this.apiHelpers = new ApiHelpers(page);
+		this.applicationsMenuPage = new ApplicationsMenuPage(page);
+		this.basePath = 'data-set-manager/entries';
+		this.dataSetsTable = page.locator('.data-set > div:nth-child(2)');
+		this.newDataSetButton = page.getByLabel('New Data Set').first();
+		this.newDataSetModal = {
+			heading: page.getByRole('heading', {name: 'New Data Set'}),
+			nameInput: page.getByLabel('NameRequired'),
+			restApplicationField: page.getByLabel('REST ApplicationRequired'),
+			restApplicationOptions: page
+				.locator('.fds-entries-dropdown-menu')
+				.first(),
+			restEndpointField: page.getByLabel('REST EndpointRequired'),
+			restEndpointOptions: page
+				.locator('.fds-entries-dropdown-menu')
+				.locator('nth=2'),
+			restSchemaField: page.getByLabel('REST SchemaRequired'),
+			restSchemaOptions: page
+				.locator('.fds-entries-dropdown-menu')
+				.locator('nth=1'),
+			saveButton: page.getByRole('button', {name: 'Save'}),
+		};
+		this.page = page;
+	}
+
+	async createTestDataSetAPI() {
+		this.apiHelpers.post(`${this.apiHelpers.baseUrl}${this.basePath}`, {
```

Comment:
Data creation shouldn't belong to the `Page`s IMO, because it is not related to the UI. Meaning you could create/need this data even from other page...

I would move it to helper classes (if not in the PR, in the future). 

In fact, once we merge the headless API, it will be even clearer that this method does not make sense in the `Page`, since the new headless API helpers don't need any page at all.

Reply:
we are going to face a pattern where we want to create some data for a bunch of tests, in the form:
* Create data in beforeall via API
* Run some tests
* Clear such data
We'll wait till such api helpers are merged to not depend on any page. In the meantime I've broken this down into smaller chunks: see  https://github.com/liferay-frontend/liferay-portal/pull/3962 and others, to facilitate merging